lib/browser/tag/update.js

```diff
@@ -160,6 +161,39 @@ function updateTagRef(expr, parent) {
   expr.tag.update()
 }
 
+/**
+ * Update dynamically created riot-tag with changing expressions
+ * @param   { Object } expr - expression tag and expression info
+ * @param   { Tag } parent - parent for tag creation
+ */
+
+function updateRtag(expr, parent) {
+  var tagName = tmpl(expr.value, parent),
+    conf
+
+  if (expr.tag && expr.tagName == tagName) {
+    expr.tag.update()
+    return
+  }
+
+  // sync _parent to accommodate changing tagnames
+  if (expr.tag) {
+    var delName = expr.tag.opts.riotTag,
+      tags = expr.tag._parent.tags[delName]
+
+    if (isArray(tags)) tags.splice(tags.indexOf(expr.tag), 1); else delete expr.tag._parent.tags[delName]
+
+  }
+
```

Comment:
@rsbondi `_parent` does not look good here because it represent __always__ the real parent of a tag child tag in a loop. Let's assume you have the following tag:
```html
<component>
   <p>{ somethingUseless }</p>
   <div each={ items }>
      <child each={ subItems } onclick={ doSomething }></child>
   </div>
</component>
```
Here the "real" `_parent` tag of all the `child` tags instances is the `<div>` element, but the `parent` exposed by the riot api is the `component`, I had to introduce it because it was needed to update only the [loop tags](https://github.com/riot/riot/blob/master/lib/browser/tag/update.js#L39). Basically clicking on a `<child>` tag will update the only the "real" parent and not the whole component. Maybe we can remove this behavior in the next riot release but it will be a big breaking change [here](https://github.com/riot/riot/issues/1319) we opened the discussion about this issue